src/bp-notifications/bp-notifications-functions.php

```diff
@@ -960,18 +960,20 @@ function ( $n ) use ( $type ) {
 	}
 
 	foreach ( $notifications as $key => $val ) {
+		
 		$user_val = get_user_meta( $user_id, $key, true );
 		if ( $user_val ) {
 			$notifications[ $key ] = $user_val;
 		}
-
-		if ( 'no' === $notifications[ $key ] && isset( $all_actions[ $key ] ) ) {
-			$excluded_actions = array_merge( $excluded_actions, $all_actions[ $key ] );
-		}
-
-		// Add in excluded action if the settings is disabled from frontend top bar Enable Notification option.
-		if ( 'no' === bp_get_user_meta( $user_id, $notifications_type_key, true ) ) {
-			$excluded_actions = array_merge( $excluded_actions, $all_actions[ $key ] );
+		
+		if (isset($all_actions[$key]) && is_array($all_actions[$key])){
+			if ( 'no' === $notifications[ $key ] ) {
+				$excluded_actions = array_merge( $excluded_actions, $all_actions[ $key ] );
+			} 		
+			// Add in excluded action if the settings is disabled from frontend top bar Enable Notification option.
+			elseif (  'no' === bp_get_user_meta( $user_id, $notifications_type_key, true ) ) {
+				$excluded_actions = array_merge( $excluded_actions, $all_actions[ $key ] );
+			}
```

Comment:
```suggestion
		if ( 'no' === $notifications[ $key ] && isset( $all_actions[ $key ] ) ) {
			$excluded_actions = array_merge( $excluded_actions, $all_actions[ $key ] );
		}

		// Add in excluded action if the settings is disabled from frontend top bar Enable Notification option.
		if ( 'no' === bp_get_user_meta( $user_id, $notifications_type_key, true ) && isset( $all_actions[ $key ] )  ) {
			$excluded_actions = array_merge( $excluded_actions, $all_actions[ $key ] );
```

@nickchomey I think we just need to add the `isset( $all_actions[ $key ] )`  with the `if ( 'no' === bp_get_user_meta( $user_id, $notifications_type_key, true ) ) {` condition.